tests/rust/tcp-test/connect/mod.rs

```diff
@@ -0,0 +1,170 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+//======================================================================================================================
+// Imports
+//======================================================================================================================
+
+use anyhow::Result;
+use demikernel::{
+    runtime::fail::Fail,
+    LibOS,
+    QDesc,
+    QToken,
+};
+use std::{
+    net::SocketAddrV4,
+    time::Duration,
+};
+
+//======================================================================================================================
+// Constants
+//======================================================================================================================
+
+#[cfg(target_os = "windows")]
+pub const AF_INET: i32 = windows::Win32::Networking::WinSock::AF_INET.0 as i32;
+
+#[cfg(target_os = "windows")]
+pub const SOCK_STREAM: i32 = windows::Win32::Networking::WinSock::SOCK_STREAM as i32;
+
+#[cfg(target_os = "linux")]
+pub const AF_INET: i32 = libc::AF_INET;
+
+#[cfg(target_os = "linux")]
+pub const SOCK_STREAM: i32 = libc::SOCK_STREAM;
+
+//======================================================================================================================
+// Standalone Functions
+//======================================================================================================================
+
+/// Drives integration tests for connect() on TCP sockets.
+pub fn run(libos: &mut LibOS, local: &SocketAddrV4, remote: &SocketAddrV4) -> Result<()> {
+    connect_invalid_queue_descriptor(libos, remote)?;
+    connect_unbound_socket(libos, remote)?;
+    connect_listening_socket(libos, local, remote)?;
```

Comment:
Should have connect_bound_socket

Reply:
Sure. I added this extra test to https://github.com/demikernel/demikernel/issues/554.